suffuse/root.go

```diff
@@ -0,0 +1,86 @@
+package suffuse
+
+import (
+  "os"
+  "bazil.org/fuse"
+)
+
+/** An interface encapsulating the essential fuse operations.
+ *  Each method returns nil if inapplicable. The fuse operations
+ *  use the first non-nil return value found for a given Path.
+ */
+type Rule interface {
+  MetaData(Path)*fuse.Attr
+  FileData(Path)[]byte
+  DirData(Path)[]fuse.Dirent
+  LinkData(Path)*Path
+}
+
+/** Eventually we'll read this kind of information out of a config
+ *  file, but for now it's hardcoded here.
+ */
+var Rules = []Rule {
+  IdRule{},  // The identity fs - passes underlying paths through unchanged
```

Comment:
There's no definitively correct order. Putting identity first means you can count on getting the actual file if there is a file with the requested name. Putting it last means any synthetic filesystem can intercept and supply something else. If there were filters, they would have to be in front of identity, but there aren't any filters yet.